apps/cowswap-frontend/src/modules/twap/utils/getTwapOrderStatus.ts

```diff
@@ -37,3 +39,53 @@ export function isTwapOrderExpired(order: TWAPOrderStruct, startDate: Date | nul
 
   return nowTimestamp > endTime
 }
+
+export function getTwapParentStatusFromChildren(
+  childrenOrders: { status: OrderStatus; isCancelling?: boolean }[],
+): TwapOrderStatus {
+  if (!childrenOrders.length) return TwapOrderStatus.Pending
+
+  const hasOpen = childrenOrders.some((order) => order.status === OrderStatus.PENDING)
+  const hasCancelling = childrenOrders.some((order) => order.isCancelling)
+  const hasFilled = childrenOrders.some((order) => order.status === OrderStatus.FULFILLED)
+  const hasCancelled = childrenOrders.some((order) => order.status === OrderStatus.CANCELLED)
+  const hasExpired = childrenOrders.some((order) => order.status === OrderStatus.EXPIRED)
+
+  // All parts filled
+  if (childrenOrders.every((order) => order.status === OrderStatus.FULFILLED)) {
+    return TwapOrderStatus.Fulfilled
+  }
+
+  // All parts cancelled
+  if (childrenOrders.every((order) => order.status === OrderStatus.CANCELLED)) {
+    return TwapOrderStatus.Cancelled
+  }
+
+  // All parts expired
+  if (childrenOrders.every((order) => order.status === OrderStatus.EXPIRED)) {
+    return TwapOrderStatus.Expired
+  }
+
+  // At least one part is open
+  if (hasOpen) {
+    return TwapOrderStatus.Pending
+  }
```

Comment:
I think this is right, but double checking that if I have

- P1: Filled
- P2: Pending

I want to show OPEN and not PARTIALLY_FILLED?